### ISSUE TITLE
Improvement: Parse Typescript built in types

### ISSUE THREAD

**Issue by jochembeetz** [SamWSoftware/serverless-auto-swagger#45]
It would be nice if the plugin would parse Typescript built in types such as: `Partial`, `Pick`, `Omit` etc.
It would make it possible to parse inferred types for example:
```
Interface Comment {
  description: string
  deletedAt: Date | null
}

Type CommentDTO = Omit<Comment, 'deletedAt'>;
```
Right now this is not possible and I would have to refactor almost all of my types, or duplicate them to have them available in Swagger.

**Comment by jochembeetz** [comment]
Opened an issue in typeconv https://github.com/grantila/typeconv/issues/20

**Comment by bfaulk96** [comment]
> Right now this is not possible and I would have to refactor almost all of my types, or duplicate them to have them available in Swagger.

Yeah, unfortunately the current solution is duplicating types. Certainly frustrating, but the way I've looked at it is I'd rather duplicate types in Typescript than have to define all my types in JSON Schema.

I upvoted your issue in typeconv!
